apps/files/lib/Controller/ViewController.php

```diff
@@ -331,6 +331,34 @@ public function index($dir = '', $view = '', $fileid = null, $fileNotFound = fal
 		$policy->addAllowedFrameDomain('\'self\'');
 		$response->setContentSecurityPolicy($policy);
 
+		$user = $this->userSession->getUser();
+		if ($openfile !== null && $user != null) {
+			$uid = $user->getUID();
+			$userFolder = $this->rootFolder->getUserFolder($uid);
+			$node = $userFolder->getById($openfile)[0];
```

Comment:
Maybe rather do something like this to properly catch invalid file ids:

```suggestion
			$nodes = $userFolder->getById($openfile);
			$node = array_shift($nodes);
			if ($node !== null) {
				// ...
			}
```